es6-shim.js

```diff
@@ -2097,6 +2170,257 @@
     // Shim incomplete iterator implementations.
     addIterator(Object.getPrototypeOf((new globals.Map()).keys()));
     addIterator(Object.getPrototypeOf((new globals.Set()).keys()));
+
+    // Reflect
+    if (!globals.Reflect) {
+      var Reflect = {};
+
+      var wrapObjectFunction = function (func) {
+        return function (target) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          try {
+            func.apply(Object, arguments);
```

Comment:
The Reflect methods should return a boolean value depending on the success/failure of an internal operation. The regular Object methods throw an error if the internal operation returns false, otherwise it returns the target object.

The advantage of this approach is that it is much shorter and simpler, but should behave the same.